addons/project/static/tests/project_task_state_selection.test.js

```diff
@@ -44,3 +44,48 @@ test("project.task (kanban): check task state widget", async () => {
         message: "When trying to click on the waiting icon, no dropdown menu should display",
     });
 });
+
+test("project.task (form): check task state widget", async () => {
+    const { env: pyEnv } = await makeMockServer();
+    ProjectTask._views = {
+        form: `<form js_class="project_task_form">
+                    <field name="project_id"/>
+                    <field name="name"/>
+                    <field name="state" widget="project_task_state_selection" nolabel="1"/>
+                </form>`,
+    };
+    const projectId = pyEnv["project.project"].create({ name: "Project One" });
+    const userId = pyEnv["res.users"].create({ name: "User one", login: "one", password: "one" });
+    const [task1, , task3] = pyEnv["project.task"].create([
+        { name: "task one", project_id: projectId, state: "01_in_progress", user_ids: [userId] },
+        { name: "task two", state: "03_approved" },
+        { name: "task three", state: "04_waiting_normal" },
+    ]);
```

Comment:
also are these needed? under HOOT we can create those records like [Project](https://github.com/odoo/odoo/blob/master/addons/project/static/tests/project_models.js#L10) .

cc @xavierbol